h5netcdf/core.py

```diff
@@ -552,9 +561,20 @@ def resize_dimension(self, dimension, size):
 class File(Group):
 
     def __init__(self, path, mode='a', invalid_netcdf=None, **kwargs):
-        self._preexisting_file = os.path.exists(path)
         try:
-            self._h5file = h5py.File(path, mode, **kwargs)
+            if path.startswith(('http', 'hdf5:')):
```

Comment:
nit: I would prefer being even more explicit here with `("http://", "https://", "hdf5://")` to match the checks in h5pyd:
https://github.com/HDFGroup/h5pyd/blob/741a8e4abacdfc0df15d9d045db0bd2cd6f47d94/h5pyd/_hl/files.py#L129